zio-http/src/main/scala/zhttp/http/Response.scala

```diff
@@ -67,12 +67,12 @@ object Response {
     headers: Headers = Headers.empty,
     data: HttpData[R, E] = HttpData.Empty,
   ): Response[R, E] = {
-    val size      = data.unsafeSize
-    val isChunked = data.isChunked
-
+    val size         = data.unsafeSize
+    val isChunked    = data.isChunked
+    val smartHeaders = if (isChunked) headers.filter(h => h._1 != Name.ContentLength) else headers
```

Comment:
Actually `TransferEncoding: Chunked` doesn't work this way. We should use `ChunkedWriter` to support this capability. The protocol is quite different than what we have been doing.